Some PEP8 Error:
======== Testing test_flake8 ========
hr_attendance_calendar/models/hr_attendance_day.py:208:39: E126 continuation line over-indented for hanging indent
hr_attendance_calendar/models/hr_attendance_day.py:209:35: E127 continuation line over-indented for visual indent
hr_attendance_calendar/models/hr_attendance_day.py:335:13: E125 continuation line with same indent as next logical line

--- a/hr_attendance_calendar/models/hr_attendance_day.py
+++ b/hr_attendance_calendar/models/hr_attendance_day.py
@@ -116,7 +116,7 @@ class HrAttendanceDay(models.Model):
                  'public_holiday_id')
     def _compute_due_hours(self):
         """First search the due hours based on the contract and after remove
-        somme hours if they are vacation"""
+        some hours if they are vacation"""
         for att_day in self:
 
             # Public holidays
@@ -125,8 +125,11 @@ class HrAttendanceDay(models.Model):
                 continue
 
             # Contract
+            # sum the due hours from today attendance prescription for the
+            # employee. These attendance are selected in
+            # update_calendar_attendance() which is called upon
+            # attendance_day creation.
             due_hours = sum(att_day.mapped('cal_att_ids.due_hours'))
-
             # Leaves
             if att_day.leave_ids:
                 for leave in att_day.leave_ids.filtered(
@@ -139,7 +142,6 @@ class HrAttendanceDay(models.Model):
 
             if due_hours < 0:
                 due_hours = 0
-
             att_day.due_hours = due_hours
 
     @api.multi
@@ -203,7 +205,8 @@ class HrAttendanceDay(models.Model):
             extra_hours = att_day.worked_hours - att_day.due_hours
             coefficient = att_day.coefficient
             att_day.extra_hours = (
-                extra_hours * coefficient) - att_day.extra_hours_lost
+                                      extra_hours * coefficient) - \
+                                  att_day.extra_hours_lost
 
     @api.multi
     def write(self, vals):
@@ -312,7 +315,6 @@ class HrAttendanceDay(models.Model):
 
         # link to schedule (resource.calendar.attendance)
         rd.update_calendar_attendance()
-
         # link to leaves (hr.holidays )
         date_str = fields.Date.to_string(att_date)
         rd.leave_ids = self.env['hr.holidays'].search([
@@ -330,10 +332,10 @@ class HrAttendanceDay(models.Model):
 
         # check public holiday
         if self.env['hr.holidays.public'].is_public_holiday(
-                rd.date, rd.employee_id.id):
+            rd.date, rd.employee_id.id):
             holidays_lines = self.env[
                 'hr.holidays.public'].get_holidays_list(
-                    att_date.year, rd.employee_id.id)
+                att_date.year, rd.employee_id.id)
             rd.public_holiday_id = holidays_lines.filtered(
                 lambda r: r.date == rd.date)
 
@@ -358,12 +360,16 @@ class HrAttendanceDay(models.Model):
         for att_day in self:
             att_date = fields.Date.from_string(att_day.date)
             week_day = att_date.weekday()
+
+            # look for a valid contract
             contracts = self.env['hr.contract'].search([
                 ('employee_id', '=', att_day.employee_id.id),
                 ('date_start', '<=', att_day.date),
                 '|', ('date_end', '=', False), ('date_end', '>=', att_day.date)
             ])
             cal_att_ids = self.env['resource.calendar.attendance']
+
+            # select the attendance(s) from the contract that is valid today.
             current_cal_att = contracts.mapped(
                 'working_hours.attendance_ids').filtered(
                 lambda a: int(a.dayofweek) == week_day)

--- a/hr_attendance_calendar/models/hr_attendance_day.py
+++ b/hr_attendance_calendar/models/hr_attendance_day.py
@@ -125,10 +125,10 @@ class HrAttendanceDay(models.Model):
                 continue
 
             # Contract
-            # sum the due hours from today attendance prescription for the
-            # employee. These attendance are selected in
-            # update_calendar_attendance() which is called upon
-            # attendance_day creation.
+            # sum the due hours extracted from the employee contract. The
+            # expected attendances are selected in update_calendar_attendance
+            # (further down in the file) which is called when an attendance
+            # day is created.
             due_hours = sum(att_day.mapped('cal_att_ids.due_hours'))
             # Leaves
             if att_day.leave_ids:
@@ -204,9 +204,8 @@ class HrAttendanceDay(models.Model):
         for att_day in self.filtered('coefficient'):
             extra_hours = att_day.worked_hours - att_day.due_hours
             coefficient = att_day.coefficient
-            att_day.extra_hours = (
-                                      extra_hours * coefficient) - \
-                                  att_day.extra_hours_lost
+            att_day.extra_hours = (extra_hours * coefficient) - \
+                att_day.extra_hours_lost
 
     @api.multi
     def write(self, vals):
@@ -332,9 +331,8 @@ class HrAttendanceDay(models.Model):
 
         # check public holiday
         if self.env['hr.holidays.public'].is_public_holiday(
-            rd.date, rd.employee_id.id):
-            holidays_lines = self.env[
-                'hr.holidays.public'].get_holidays_list(
+                rd.date, rd.employee_id.id):
+            holidays_lines = self.env['hr.holidays.public'].get_holidays_list(
                 att_date.year, rd.employee_id.id)
             rd.public_holiday_id = holidays_lines.filtered(
                 lambda r: r.date == rd.date)

--- a/hr_attendance_calendar/models/hr_employee.py
+++ b/hr_attendance_calendar/models/hr_employee.py
@@ -76,9 +76,19 @@ class HrEmployee(models.Model):
         employees = self.search([])
         today = fields.Date.today()
         for employee in employees:
+            # check if an entry already exists. If yes, it will not be
+            # recreated
             att_days = att_day.search(
-                [('name', '=', today), ('employee_id', '=', employee.id)])
-            if not att_days:
+                [('date', '=', today), ('employee_id', '=', employee.id)])
+
+            # check that the employee is currently employed.
+            contracts_valid_today = self.env['hr.contract'].search([
+                ('employee_id', '=', employee.id),
+                ('date_start', '<=', today),
+                '|', ('date_end', '=', False), ('date_end', '>=', today)
+            ])
+
+            if not att_days and contracts_valid_today:
                 att_day.create({
                     'date': today,
                     'employee_id': employee.id

--- a/hr_attendance_calendar/views/hr_attendance_day_view.xml
+++ b/hr_attendance_calendar/views/hr_attendance_day_view.xml
@@ -78,9 +78,10 @@
                 <field name="date"/>
                 <field name="working_day"/>
                 <field name="employee_id"/>
-                <field name="worked_hours" widget="float_time"/>
-                <field name="extra_hours" widget="float_time"/>
-                <field name="extra_hours_lost" widget="float_time"/>
+                <field name="worked_hours" widget="float_time" type="char"/>
+                <field name="extra_hours_lost" widget="float_time"
+                       type="char"/>
+                <field name="extra_hours" widget="float_time" type="char"/>
             </tree>
         </field>
     </record>

